### ISSUE TITLE
[FE] - Accessibility checks required for staging review - Axe scans in daily development - Screenreader test

### ISSUE THREAD

**Issue by mshea0606** [department-of-veterans-affairs/va.gov-team#11273]
## Background
There are a few 508/accessibility checks required as a pre-requisite for the staging review. They are listed below.

## Tasks
- [x] [Axe scans in daily development](https://github.com/department-of-veterans-affairs/va.gov-team/blob/master/platform/accessibility/guidance/staging-review-processes.md#axe-scans-in-daily-development) 
- [x] [Screen reader tests](https://github.com/department-of-veterans-affairs/va.gov-team/blob/master/platform/accessibility/guidance/staging-review-processes.md#screen-reader-tests)

**Comment by sanlouise** [comment]
Notes:

**Axe Scans**
(1) 5x warning for 'Elements must have sufficient color contrast' on elements that were not altered during the development of Profile 2.0. (See screenshot)
(2) A warning saying that `role="presentation"` is not the correct attribute on the service badge in `ProfileHeader`, so I removed that attribute in [this PR](https://github.com/department-of-veterans-affairs/vets-website/pull/13516/files)

![image](https://user-images.githubusercontent.com/14869324/88080717-a8e0c300-cb3c-11ea-9631-d7efcfc40497.png)

**Comment by sanlouise** [comment]
Notes:

**Personal and contact information**

1. The screen reader does not read all content on this alert (edit a field and then try to navigate away from the page)
![image](https://user-images.githubusercontent.com/14869324/88218109-cbe2a400-cc1c-11ea-9aa4-f0b12b4cdae7.png)

2. When having a section in an editing state, and then when trying to edit another field, this modal pops up. Using a screen reader, I found it hard to navigate back to the section I was editing. I wonder if we need a button directly on this modal that allows the user to 'cancel' out of the section they were editing to ensure fewer steps. 
![image](https://user-images.githubusercontent.com/14869324/88218438-4d3a3680-cc1d-11ea-95d2-bff24d61bd75.png)

**Connected apps**
1. The screen reader says 'Black square', should we mute this?
![image](https://user-images.githubusercontent.com/14869324/88218593-79ee4e00-cc1d-11ea-995c-842e8e5441b6.png)


**General**
1. The forms 'Update' and 'Cancel' button are not descriptive to screen readers 
![image](https://user-images.githubusercontent.com/14869324/88218536-6cd15f00-cc1d-11ea-80dc-21624ed9be77.png)

2. The 'Your session will end in...' modal is read by the screen reader, but the action buttons can be hidden underneath another modal such as the 'Disconnect App' modal in Connected apps or the 'You're currently editing your ... section' in Personal and contact information. This makes it hard for the user to indicate that they need more time.

**Comment by jenstrickland** [comment]
> Notes:
> 
> **Axe Scans**
> (1) 5x warning for 'Elements must have sufficient color contrast' on elements that were not altered during the development of Profile 2.0. (See screenshot)
> (2) A warning saying that `role="presentation"` is not the correct attribute on the service badge in `ProfileHeader`, so I removed that attribute in [this PR](https://github.com/department-of-veterans-affairs/vets-website/pull/13516/files)
> 
> ![image](https://user-images.githubusercontent.com/14869324/88080717-a8e0c300-cb3c-11ea-9631-d7efcfc40497.png)

Teams only need to scan their project's scope. 

1. The background color manual checks shown need to be looked at manually. Since these exist in the header, footer, and select elements, they are not an issue. For these, select the radio input that says, "this is not an issue."
1. Regarding `role="presentation"` warning, that happened because there is alt text for the military branch seal, so removing the role is the right action.

**Comment by jenstrickland** [comment]
> Screen reader Notes:
> 
> **Personal and contact information**
> 
> 1. The screen reader does not read all content on this alert (edit a field and then try to navigate away from the page)
>    ![image](https://user-images.githubusercontent.com/14869324/88218109-cbe2a400-cc1c-11ea-9aa4-f0b12b4cdae7.png)



I tested this and it's fine. It read all the info on the alert, starting with the button that has focus.

> 2. When having a section in an editing state, and then when trying to edit another field, this modal pops up. Using a screen reader, I found it hard to navigate back to the section I was editing. I wonder if we need a button directly on this modal that allows the user to 'cancel' out of the section they were editing to ensure fewer steps.
>    ![image](https://user-images.githubusercontent.com/14869324/88218438-4d3a3680-cc1d-11ea-95d2-bff24d61bd75.png)



The Close button allows the user to cancel the action. I appreciate empathizing with how awkward it is. :)


> **Connected apps**
> 
> 1. The screen reader says 'Black square', should we mute this?
>    ![image](https://user-images.githubusercontent.com/14869324/88218593-79ee4e00-cc1d-11ea-995c-842e8e5441b6.png)
> 


I would love to mute the black square, and that's a global concern and already raised with the design system folks.


> **General**
> 
> 1. The forms 'Update' and 'Cancel' button are not descriptive to screen readers
>    ![image](https://user-images.githubusercontent.com/14869324/88218536-6cd15f00-cc1d-11ea-80dc-21624ed9be77.png)

The current 'Update' and 'Cancel' behavior is a global pattern and we should maintain consistency.  @1Copenut for visibility, as this may be an enhancement to consider, that we dynamically add sr-only text to clarify what is being updated or canceled.

> 2. The 'Your session will end in...' modal is read by the screen reader, but the action buttons can be hidden underneath another modal such as the 'Disconnect App' modal in Connected apps or the 'You're currently editing your ... section' in Personal and contact information. This makes it hard for the user to indicate that they need more time.

This issue is of consequence. We need to set a z-index order so that the "Your session will end in… " modal always appears on top of everything else. @1Copenut — this would be a sitewide Formation issue, correct?
